src/main/java/com/exasol/adapter/dynamodb/documentpath/DocumentPathExpression.java

```diff
@@ -75,6 +75,36 @@ public String toString() {
         return new DocumentPathToStringConverter().convertToString(this);
     }
 
+    /**
+     * Gives the index of the first {@link ArrayAllPathSegment}.
+     *
+     * @return index of the first {@link ArrayAllPathSegment}; {@code -1} if not found.
+     */
+    int indexOfFirstArrayAllSegment() {
+        int index = 0;
+        for (final PathSegment pathSegment : this.path) {
+            if (pathSegment instanceof ArrayAllPathSegment) {
+                return index;
+            }
+            index++;
+        }
+        return -1;
+    }
+
+    /**
+     * Checks if this path expression starts with an other path expression.
+     * 
+     * @param other sub path expression
+     * @return {@code true} if this path starts with the other path
+     */
+    public boolean startsWith(final DocumentPathExpression other) {
+        if (other.size() > this.size()) {
+            return false;
+        }
+        final DocumentPathExpression thisSubpathWithLengthOfOther = this.getSubPath(0, other.size());
+        return thisSubpathWithLengthOfOther.equals(other);
```

Comment:
I checked the equals method of the class, but github does not allow to change it (outside of the changes in PR).

Please change it to use `isInstanceOf` instead of `getClass()`. Lines [59-61](https://github.com/exasol/dynamodb-virtual-schema/blob/ee3f41d69f2f52df95970bdab933c554ca21fec4/src/main/java/com/exasol/adapter/dynamodb/documentpath/DocumentPathExpression.java#L59-L61):

```java
        if (!(other instanceof DocumentPathExpression)) {
            return false;
        }
```

This is general good practice. Using `getClass` can break the [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle). You can read more in Effective Java, Item 10.

Reply:
thank you for the hint. I changed it forall my equals implementations. This was the default of intelij. I should probably change the default ;-)